File: AtCoder/ACP/BootCamp4b/Hard100_2/python/C_String_Transformation.py
S = input()
T = input()
CS = {}
CT = {}
f = True
for i in range(len(S)):
    if S[i] in CS and CS[S[i]] != T[i]:
        f = False
        break
    else:
        CS[S[i]] = T[i]
    if T[i] in CT and CT[T[i]] != S[i]:
        f = False
        break
    else:
        CT[T[i]] = S[i]
if f:
    print("Yes")
else:
    print("No")

File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/C_String_Transformation.cpp
#include <bits/stdc++.h>
using namespace std;

string S, T;
char CS[26];
char CT[26];

void solve() {
  bool f = true;
  for (int i = 0; i < S.size(); ++i) {
    if (CS[S[i] - 'a'] && CS[S[i] - 'a'] != T[i]) {
      f = false;
      break;
    } else {
      CS[S[i] - 'a'] = T[i];
    }
    if (CT[T[i] - 'a'] && CT[T[i] - 'a'] != S[i]) {
      f = false;
      break;
    } else {
      CT[T[i] - 'a'] = S[i];
    }
  }
  if (f) {
    puts("Yes");
  } else {
    puts("No");
  }
}

int main() {
  cin >> S >> T;
  solve();
}